src/App.tsx

```diff
@@ -42,54 +66,102 @@ export function getReorderedGoods(
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
+
+  reverse = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortAlphabetically = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const goods = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+    const startOrder = sortType === SortType.NONE && isReversed === false;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            onClick={this.sortAlphabetically}
+            type="button"
+            className={`button is-info ${sortType === SortType.ALPHABET
+              ? ''
+              : 'is-light'
+            }`}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            onClick={this.sortByLength}
+            type="button"
+            className={`button is-success ${sortType === SortType.LENGTH
+              ? ''
+              : 'is-light'
+            }`}
+          >
+            Sort by length
+          </button>
+
+          <button
+            onClick={this.reverse}
+            type="button"
+            className={`button is-warning ${isReversed
+              ? ''
+              : ' is-light'
+            }`}
+          >
+            Reverse
+          </button>
+          {!startOrder && (
+            <button
+              onClick={this.reset}
+              type="button"
+              className="button is-danger is-light"
+            >
+              Reset
+            </button>
+          )}
+        </div>
 
-      <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          <ul>
+            {goods.map(good => (
+              <li
+                key={good}
+                data-cy="Good"
+              >
+                {good}
+              </li>
+            ))}
+          </ul>
         </ul>
```

Comment:
Hm, there is an `<ul>` inside another `<ul>`. 
Nested `<ul>` tags can lead to incorrect rendering and potential display issues in web browsers. Therefore, it is best to remove the nested `<ul>` and ensure that you have a proper HTML structure.